systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAbstractST.java

```diff
@@ -71,73 +70,62 @@ void setup() throws Exception {
-                                        new CertSecretSourceBuilder()
-                                            .withSecretName(SECRET_OF_KEYCLOAK)
-                                            .withCertificate(CERTIFICATE_OF_KEYCLOAK)
-                                            .build())
-                                    .withDisableTlsHostnameVerification(true)
-                                .endKafkaListenerAuthenticationOAuth()
-                            .endKafkaListenerExternalNodePort()
-                        .endListeners()
-                    .endKafka()
-                .endSpec()
-                .done();
+            .editSpec()
+                .editKafka()
+                    .editListeners()
+                        .withNewTls()
+                            .withNewKafkaListenerAuthenticationOAuth()
+                                .withValidIssuerUri(keycloakInstance.getValidIssuerUri())
+                                .withJwksExpirySeconds(keycloakInstance.getJwksExpireSeconds())
+                                .withJwksRefreshSeconds(keycloakInstance.getJwksRefreshSeconds())
+                                .withJwksEndpointUri(keycloakInstance.getJwksEndpointUri())
+                                .withUserNameClaim(keycloakInstance.getUserNameClaim())
+                                .withTlsTrustedCertificates(
+                                    new CertSecretSourceBuilder()
+                                        .withSecretName(SECRET_OF_KEYCLOAK)
+                                        .withCertificate(CERTIFICATE_OF_KEYCLOAK)
+                                        .build())
+                                .withDisableTlsHostnameVerification(true)
+                            .endKafkaListenerAuthenticationOAuth()
+                        .endTls()
+                        .withNewKafkaListenerExternalNodePort()
+                            .withNewKafkaListenerAuthenticationOAuth()
+                                .withValidIssuerUri(keycloakInstance.getValidIssuerUri())
+                                .withJwksExpirySeconds(keycloakInstance.getJwksExpireSeconds())
+                                .withJwksRefreshSeconds(keycloakInstance.getJwksRefreshSeconds())
+                                .withJwksEndpointUri(keycloakInstance.getJwksEndpointUri())
+                                .withUserNameClaim(keycloakInstance.getUserNameClaim())
+                                .withTlsTrustedCertificates(
+                                    new CertSecretSourceBuilder()
+                                        .withSecretName(SECRET_OF_KEYCLOAK)
+                                        .withCertificate(CERTIFICATE_OF_KEYCLOAK)
+                                        .build())
+                                .withDisableTlsHostnameVerification(true)
+                            .endKafkaListenerAuthenticationOAuth()
+                        .endKafkaListenerExternalNodePort()
```

Comment:
Do we still need this?

Reply:
Yes because I didn't removed external listener completely. I expected it will be done in our PR